mirrord/protocol/src/codec.rs

```diff
@@ -142,77 +143,37 @@ pub enum DaemonMessage {
     SwitchProtocolVersionResponse(#[bincode(with_serde)] semver::Version),
 }
 
-pub struct ClientCodec {
+pub struct ProtocolCodec<I, O> {
```

Comment:
Why not require `I: Decode` and `O: Encode`?

Reply:
Alright. I'm just a fan of type constraints, and here we always have this constraint.

But it's fine to leave it out, we don't interact with this type much anyway.